File: pubnub-gson/src/main/java/com/pubnub/api/endpoints/presence/GetState.java
package com.pubnub.api.endpoints.presence;

import com.pubnub.api.endpoints.Endpoint;
import com.pubnub.api.models.consumer.presence.PNGetStateResult;

public interface GetState extends Endpoint<PNGetStateResult> {
    GetState channels(java.util.List<String> channels);

    GetState channelGroups(java.util.List<String> channelGroups);

    GetState uuid(String uuid);
}

File: pubnub-kotlin/pubnub-kotlin-api/src/jvmMain/kotlin/com/pubnub/api/endpoints/presence/GetState.kt
package com.pubnub.api.endpoints.presence

import com.pubnub.api.Endpoint
import com.pubnub.api.models.consumer.presence.PNGetStateResult

/**
 * @see [PubNub.getPresenceState]
 */
actual interface GetState : Endpoint<PNGetStateResult> {
    val channels: List<String>
    val channelGroups: List<String>
    val uuid: String
}
